### PR TITLE
SecuritySuite 3.1.16

### PR DESCRIPTION
**Pull request by eggsurplus**
This brings SecuritySuite up to 3.1.16 on all bug fixes.

## Description
Includes fixes for the blank rows in the My Activity Stream, changing over to static functions as appropriate, and group selector on new record creation.

## Motivation and Context
Long overdue and also due to this issue: #1491

## How To Test This
Do groups still work?
Is the blank row problem solved?

## Types of changes
- [X] Bug fix (non-breaking change which fixes an issue)

### Final checklist
- [X] My code follows the code style of this project found [here](https://docs.suitecrm.com/community/contributing-code/coding-standards/).
- [X] I have read the [**How to Contribute**](https://docs.suitecrm.com/community/contributing-code/) guidelines.

### CODE REVIEW

**Comment by Dillon-Brown** [comment]
Hi @eggsurplus, it looks like this change causes the installer to fail when attempting to create the database. Is this something you would be able to look into?

**Comment by eggsurplus** [comment]
@Dillon-Brown Just tried a clean install with demo data and it appears to work just fine. Do you have any more information on the case where the installer failed for you? The changes shouldn't have any impact on the installer, but I've seen crazier things happen before.

**Comment by Dillon-Brown** [comment]
@eggsurplus Problem seems to be in ```modules/SecurityGroups/SecurityGroup.php```. This is where Travis is failing: 
bpVTM2UserWizardCest testScenarioInstallSuiteCRMWithDefaultConfiguration fail

**Comment by eggsurplus** [comment]
@Dillon-Brown Thanks, this is what I needed. Will look into it. Switching some non-static functions over to static as it should be and likely missed something during the process.

**Comment by eggsurplus** [comment]
@Dillon-Brown Fixed and committed. 

https://github.com/salesagility/SuiteCRM/pull/7474/commits/fa76f16cfd7fa2979da6c8f693bf7d1f2db1721e

**Comment by Dillon-Brown** [comment]
@eggsurplus Few more issues with static functions. This commit appears to resolve the problem: https://github.com/Dillon-Brown/SuiteCRM/commit/ea0e840e069784814938242734b3ad3f54fd6911

File: modules/SecurityGroups/SecurityGroup.php [hunk 1]
```diff
@@ -792,4 +793,77 @@ public static function getPrimaryGroupID()
 
         return $primary_group_id;
     }
+
+    //used in EditView2 to figure out what the parent security groups are set to
+    static function getParentGroups($focus)
```
Comment (gymad) [inline modules/SecurityGroups/SecurityGroup.php]: missing visibility

File: modules/SecurityGroups/SecurityGroup.php [hunk 2]
```diff
@@ -792,4 +793,77 @@ public static function getPrimaryGroupID()
...
+                $parent_bean->load_relationship($rel_name);
+                if(!empty($parent_bean->$rel_name))
+                {
+                    $groups = $parent_bean->$rel_name->getBeans();
+                    //reorganize to index by id
+                    if(!empty($groups))
+                    {
+                        foreach($groups as $group)
+                        {
+                            $parent_groups[$group->id] = $group;
```
Comment (gymad) [inline modules/SecurityGroups/SecurityGroup.php]: extra large complexity

File: include/EditView/EditView2.php [hunk 3]
```diff
@@ -932,46 +932,83 @@ public function display($showTitle = true, $ajaxSave = false)
...
-
+                    $ss_mod_strings = return_module_language($current_language, 'SecurityGroups');  
+                    
                     $lbl_securitygroups_select = $ss_mod_strings['LBL_GROUP_SELECT'];
                     $lbl_securitygroups = $ss_mod_strings['LBL_LIST_FORM_TITLE'];
+                    
+                    //SuiteCRM: check which theme this is for.
+                    if($theme == 'SuiteP' || $theme == 'SuitePImproved')
+                    {
+                        $group_panel = <<<EOQ
```
Comment (gymad) [inline include/EditView/EditView2.php]: Maybe better practice is using a template engine. In SuiteCRM we using smarty.

**Comment by Mac-Rae** [comment]
Assessed :+1:

**Comment by eggsurplus** [comment]
If someone wants to alter the commit go for it. I've put in more time than I have for this. Just wanted to contribute the fixes that are in the full version of SecuritySuite on The SuiteCRM Store.

**Comment by Dillon-Brown** [comment]
Hi @eggsurplus, no worries, I can take over this PR until it gets merged. Thanks!

**Pull request merged**